cmake/AmiciFindBLAS.cmake

```diff
@@ -12,18 +12,32 @@ set_property(CACHE BLAS PROPERTY STRINGS "CBLAS" "MKL" "ACCELERATE")
 
 if(${BLAS} STREQUAL "MKL" OR DEFINED ENV{MKLROOT})
   if(DEFINED ENV{MKLROOT})
-    # This is set by Environment Modules
-    message(STATUS "Using MKL_INCDIR and MKL_LIB from environment module")
     set(BLAS
         "MKL"
         CACHE STRING "BLAS library to use" FORCE)
-    set(BLAS_INCLUDE_DIRS
-        "$ENV{MKL_INCDIR}"
-        CACHE STRING "" FORCE)
-    set(BLAS_LIBRARIES
-        "$ENV{MKL_LIB}"
-        CACHE STRING "" FORCE)
+
+    # Was MKLROOT set by /opt/intel/oneapi/setvars.sh? then cmake will find it
+    message(STATUS "Trying to find BLAS based on MKLROOT=$ENV{MKLROOT}")
+    # give the user the option to override the BLA_VENDOR
+    if(NOT DEFINED BLA_VENDOR)
+      set(BLA_VENDOR Intel10_64lp)
+    endif()
+    find_package(BLAS)
```

Comment:
Is this really only going to work if `MKLROOT` is set?

Reply:
`find_package(BLAS)`? No, generally MKLROOT does not need to be set. But to use the Intel MKL, it is most likely required. I'd have to check the full cmake module to see which other environment variables might be considered, but with a default installation, it only works after setting certain env variables, and MKLROOT is one of them.

The `find_package(BLAS)` case without MKLROOT is handled further down.

Reply:
ah, missed the `if(DEFINED ENV{MKLROOT})`